platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -236,7 +236,10 @@ odp_crypto_alg_err_t aes_gcm_encrypt(odp_packet_t pkt,
 	 * read it in 16 byte chunks. */
 	uint8_t aad[ARM_CRYPTO_MAX_AAD_LENGTH];
 
-	memcpy(aad, param->aad_ptr, session->p.auth_aad_len);
+	if (session->p.auth_aad_len == 8)
+		memcpy(aad, param->aad_ptr, 8);
```

Comment:
Alternatively:
```C
memcpy(aad, param->aad_ptr, session->p.auth_aad_len == 8 ? 8 : 12);
```

Reply:
OK, this could be in the commit (cbcad2c) log.

Reply:
Improved the log in v4 and also moved the optimized copy to a function and added a comment in the code.